ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/hooks/useDeleteModal.ts

```diff
@@ -0,0 +1,57 @@
+import useModal from 'hooks/useModal';
+import { useState } from 'react';
+import { deleteReportConfiguration } from 'services/ReportsService';
+import { getErrorMessage } from '../errorUtils';
+
+export type UseDeleteModalResult = {
+    openDeleteModal: (reportId: string) => void;
+    isDeleteModalOpen: boolean;
+    closeDeleteModal: () => void;
+    isDeleting: boolean;
+    onDelete: () => void;
+    deleteError: string | null;
+    isDeleted: boolean;
+};
+
+function useDeleteModal(): UseDeleteModalResult {
+    const { isModalOpen: isDeleteModalOpen, openModal, closeModal } = useModal();
+    const [reportIdToDelete, setReportIdToDelete] = useState<string>('');
+    const [isDeleting, setIsDeleting] = useState(false);
+    const [isDeleted, setIsDeleted] = useState(false);
+    const [deleteError, setDeleteError] = useState<string | null>(null);
+
+    function openDeleteModal(reportId: string) {
+        openModal();
+        setReportIdToDelete(reportId);
+    }
```

Comment:
It seems like this function should also reset the states back to their defaults. e.g. On the list page if you delete a report and then try to delete a second report, both deletion flows will use this same hook. The values of `isDeleted`, `isDeleting`, and `deleteError` will be stale from the last deletion and might cause issues if they are relied on elsewhere.